### PR TITLE
feat: Add support for wasm module imports in `wrangler pages dev`

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**
**What this PR solves / how to test:**
tldr 👉  `wrangler pages dev <directory> --experimental-worker-bundle=true`

Currently it is not possible to import `wasm` modules in either Pages
Functions or Pages Advanced Mode projects.

This commit caries out work to address the aforementioned issue by
enabling `wasm` module imports in `wrangler pages dev`. As a result,
Pages users can now import their `wasm` modules withing their Functions
or `_worker.js` files, and `wrangler pages dev` will correctly bundle
everything and serve these "external" modules.

```
import hello from "./hello.wasm"

export async function onRequest() {
	const module = await WebAssembly.instantiate(hello);
	return new Response(module.exports.hello);
}
```

depends on https://github.com/cloudflare/wrangler2/pull/2563

**Associated docs issues/PR:**
N/A

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset

**Reviewer has performed the following, where applicable:**

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Fixes # [insert issue number].

### CODE REVIEW

File: fixtures/pages-functions-wasm-app/wasm/add.wat [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(module
```
Comment (CarmenPopoviciu) [inline fixtures/pages-functions-wasm-app/wasm/add.wat:1]: `add.wasm` is generated from this file via `wat2wasm add.wat -o addwasm `. Technically this file is not needed, but wanted to keep it for readability purposes

File: fixtures/pages-functions-wasm-app/public/index.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE html>
```
Comment (petebacondarwin) [inline fixtures/pages-functions-wasm-app/public/index.html:1]: Is this needed?

Comment (CarmenPopoviciu) [reply fixtures/pages-functions-wasm-app/public/index.html:1]: the index file you mean?

File: fixtures/pages-functions-wasm-app/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# ⚡️ pages-functions-wasm-app
+
+`pages-functions-wasm-app` is a test fixture that sets up a ⚡️Pages project with [Functions](https://developers.cloudflare.com/pages/platform/functions) and [`wasm` module imports](https://blog.cloudflare.com/workers-javascript-modules/)
+
+## Dev
+
+```bash
+# cd into the test fixture folder
+cd fixtures/pages-functions-with-routes-app
```
Comment (GregBrimble) [inline fixtures/pages-functions-wasm-app/README.md:9]:
```suggestion
cd fixtures/pages-functions-wasm-app
```

File: fixtures/pages-functions-wasm-app/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+npm run dev
+```
+
+## Publish
+
+> Please note that in order to deploy this project to `.pages.dev` you need to have a [Cloudflare account](https://dash.cloudflare.com/login)
+
+```bash
+# cd into the test fixture folder
+cd fixtures/pages-functions-with-routes-app
```
Comment (GregBrimble) [inline fixtures/pages-functions-wasm-app/README.md:21]:
```suggestion
cd fixtures/pages-functions-wasm-app
```

File: fixtures/pages-functions-wasm-app/wasm/add.wat [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  (func $add (param $p1 i32) (param $p2 i32) (result i32)
+    local.get $p1
+    local.get $p2
+    i32.add)
+  (export "add" (func $add))
+)
```
Comment (GregBrimble) [inline fixtures/pages-functions-wasm-app/wasm/add.wat:7]:
```suggestion
)

```

File: fixtures/pages-workerjs-wasm-app/wasm/multiply.wat [hunk 6]
```diff
@@ -0,0 +1,7 @@
+(module
+  (func $multiply (param $p1 i32) (param $p2 i32) (result i32)
+    local.get $p1
+    local.get $p2
+    i32.mul)
+  (export "multiply" (func $multiply))
+)
```
Comment (GregBrimble) [inline fixtures/pages-workerjs-wasm-app/wasm/multiply.wat:7]:
```suggestion
)

```

**Review by GregBrimble — APPROVED** [review]
Woo! We're almost done! :)

File: fixtures/pages-functions-wasm-app/README.md (hunk 3 above)
Comment (CarmenPopoviciu) [reply fixtures/pages-functions-wasm-app/README.md:9]: lol...I double checked so many times, and I still missed some of these. Nice catch!

**Comment by CarmenPopoviciu** [comment]
@petebacondarwin anything you want to add? or can I go ahead and merge?

**Comment by petebacondarwin** [comment]
Sorry @CarmenPopoviciu - I meant to say it was fine by me.
